src/registries/autofill_rules.ts

```diff
@@ -154,3 +253,97 @@ autofillRulesRegistry
     },
     sequence: 40,
   });
+
+function getDateIntervals(dates: DateTime[]): DateIncrement[] {
+  if (dates.length < 2) {
+    return [{ years: 0, months: 0, days: 0 }];
+  }
+
+  const res = dates.map((date, index) => {
+    if (index === 0) {
+      return { years: 0, months: 0, days: 0 };
+    }
+    const previous = dates[index - 1];
+    const years = yearsBetween(previous, date);
+    const months = monthsBetween(previous, date);
+    previous.setFullYear(previous.getFullYear() + years);
+    previous.setMonth(previous.getMonth() + months);
+    const days = daysBetween(previous, date);
+
+    return {
+      years,
+      months,
+      days,
+    };
+  });
+  return res.slice(1);
+}
+
+function daysBetween(date1: DateTime, date2: DateTime) {
```

Comment:
We already have helpers for the difference in days/month/years between dates (DATEDIF function)

Reply:
I figured but  failed to find them >< thanks :)